src/osd/OSD.cc

```diff
@@ -10556,6 +10557,8 @@ void OSD::ShardedOpWQ::_process(uint32_t thread_index, heartbeat_handle_d *hb)
     }
   }
 
+  sdata->shard_in_progress = true;
```

Comment:
you're changing this condition without protecting it with `sdata->sdata_wait_lock`.

i think this is the reason the osd_repop requests were not handled and replied in 5 minutes. see 

- http://pulpito.ceph.com/kchai-2019-08-16_05:52:53-rados-wip-kefu-testing-2019-08-16-0928-distro-basic-smithi/4220753/
- http://pulpito.ceph.com/kchai-2019-08-15_14:51:08-fs-wip-kefu-testing-2019-08-15-1951-distro-basic-smithi/


also, could you paste the benchmark results before and after this change?